rust/xaynet-core/src/mask/masking.rs

```diff
@@ -585,6 +587,155 @@ mod tests {
     test_masking!(pow_i64_b6, Power2, i64, 1_000_000, 10);
     test_masking!(pow_i64_bmax, Power2, i64, 10);
 
+    /// Generate tests for masking and unmasking of a single model:
+    /// - generate random scalar from a uniform distribution with a seeded PRNG
+    /// - scale a model of unit weights and mask it
+    /// - check that all masked weights belong to the chosen finite group
+    /// - unmask the masked model
+    /// - check that all unmasked weights are equal to the original weights (up to a tolerance
+    ///   determined by the masking configuration)
+    ///
+    /// The arguments to the macro are:
+    /// - a suffix for the test name
+    /// - the group type of the model (variants of `GroupType`)
+    /// - the data type of the model (either primitives or variants of `DataType`)
+    /// - an absolute bound for the weights (optional, choices: 1, 100, 10_000, 1_000_000)
```

Comment:
we actually need two data types here, one for the model as usual and one for the scalar, because in our masking setup the scalar must always be a floating point value (which would not hold for integer models). since the model is set to constant 1 for this test, we down gain anything from having int and float data types for the model and we might simplify a bit and only call this macro with float data types, eliminating the need to have two different data types.

since we currently go for the same mask configuration for both model and scalar, we probably should take advantage of this simplification and extend the test later on.

```suggestion
    /// The arguments to the macro are:
    /// - a suffix for the test name
    /// - the group type of the model and scalar (variants of `GroupType`)
    /// - the data type of the model and sclar (either float primitives or float variants of `DataType`)
    /// - an absolute bound for the scalar (optional, choices: 1, 100, 10_000, 1_000_000)
```

Reply:
not sure I follow, are you saying that the `i32`/`i64` cases are redundant here?

Reply:
what i mean is that in the current implementation the values for the scalar are drawn from the same data type as the model. in case of `i32`/`i64` the random scalars would only take integer values, which doesn't make much sense regarding the definition and purpose of the scalar.

Reply:
I see, so I think you're saying that in practice, one wouldn't actually take scalars to be int values. that may well be the case, but there is no harm as such in allowing for this, is there?

Reply:
there is probably no harm, it just doesn't make much sense to test for sth that isn't intended in the first place.

Reply:
fair enough. we can consider bringing them back then once we can properly mix & match mask configurations.